Update and improve minimum system requirements

milljm opened the pull request:
Update the minimum stated gcc version

Closes #27449

Commit:
Bump the min gcc

Update the minimum stated gcc version

Closes #27449

Commit:
Bump GCC min version

Make this information more visiable by adding it to the front installation
page.

Closes #27449

Commit:
Convert to a table

Convert the list of requirements to a table.

Add more template control versions to packages_config.yml. Some are not
being used, but I figured they may come in handy.

cticenhour commented on framework/doc/content/sqa/minimum_requirements.md:
```diff
@@ -4,14 +4,21 @@
 !style-end!
 
-- GCC/Clang C++17 compliant compiler (GCC @ [!package!minimum_gcc], Clang @ [!package!minimum_clang] or greater)
+A [!ac](POSIX) compliant Unix-like operating system. Any modern Linux operating system, or a
+Macintosh machine running either of the last two releases.
```
```suggestion
A [!ac](POSIX) compliant Unix-like operating system. This includes any modern Linux-based operating system (e.g., Ubuntu, Fedora, Rocky, etc.), or a
Macintosh machine running either of the last two MacOS releases.
```

I still like calling them "Macintosh" too 😂

cticenhour commented on framework/doc/content/sqa/minimum_requirements.md:
```diff
@@ -4,14 +4,21 @@
+| :- | :- |
+| GCC | [!package!minimum_gcc] - [!package!maximum_gcc] |
+| LLVM/Clang | [!package!minimum_clang] - [!package!maximum_clang] |
+| Intel | Not supported at this time |
```
```suggestion
| Intel (ICC / ICX) | Not supported at this time |
```

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2190938/) on 63d8c64 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27454/site)

This comment will be updated on new commits.

Commit:
re-word a few things

Add more to the description of what a modern linux operating system
is.

Add compiler binary names for Intel.

Drop the abbreviation for Pkgs since were going to have a new length
anyways (i am not going to try to align this up again, lets just get
this in).

Closes #27449
